src/global/overlay/overlay.tsx

```diff
@@ -16,3 +18,101 @@ export function overlayGapFixCorners(): JSX.Element {
     </div>,
   ];
 }
+
+/**
+ * Returns all the ancestors of the overlay.
+ */
+const getAncestors = (overlay: HTMLElement, root: HTMLElement): HTMLElement[] => {
+  let el = overlay.parentElement;
+  const ancestors: HTMLElement[] = [];
+  while (el) {
+    if (el.matches(IS_OPEN_OVERLAY_QUERY)) {
```

Comment:
Is it intentional, that this logic also sets sbbInert? Because this way it seems to be done twice; Once here and once afterwards on line 96.

Reply:
Ah, I think I understand now, but it's still a bit confusing. Could this maybe be done in `applyInertMechanism`?

Reply:
Since it is an ancestor, the line 96 should not apply to it, this set only the custom attr to a partent that is an overlay. I needed that for the nested overlays case and to avoid a NaN on the line 106. But I forgot to remove it as for now we don't support the nested case, so I will remove it.

Reply:
Or should I leave it to at least partially cover the scenario (except the shadowRoot obvs)?